packages/logging/src/log.js

```diff
@@ -136,6 +137,27 @@ Log.formatName_ = function(projectId, name) {
   return path + name;
 };
 
+Log.prototype.assignDefaultMetadata_ = function(entry, callback) {
+  var entryMetadata = entry.metadata;
+  // If the resource specifcally was not supplied attempt to determine it
+  // automatically
+  if (!is.object(entryMetadata) || !is.object(entryMetadata.resource)) {
+    this.metadata_.getDefaultResource(function(err, resource) {
+      if (err) {
+        callback(err);
```

Comment:
Should we just continue making the request with the object as it was provided by the user, instead of returning an error because we were unable to decorate it? Presumably, if something was required upstream that we were unable to find, the upstream API would have its own error to communicate this, that we would then provide to the user.